server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -1043,7 +1080,7 @@ public IndexSettings(final IndexMetadata indexMetadata, final Settings nodeSetti
 
         setEnableFuzzySetForDocId(scopedSettings.get(INDEX_DOC_ID_FUZZY_SET_ENABLED_SETTING));
         setDocIdFuzzySetFalsePositiveProbability(scopedSettings.get(INDEX_DOC_ID_FUZZY_SET_FALSE_POSITIVE_PROBABILITY_SETTING));
-
+        isCompositeIndex = scopedSettings.get(StarTreeIndexSettings.IS_COMPOSITE_INDEX_SETTING);
```

Comment:
1. We expects users to pass this, right?
2. Can we have a validation that mapping for composite field can only be set if this is true?

Reply:
1. Yes
2. Yes the validation is added already - refer https://github.com/opensearch-project/OpenSearch/pull/15533/files#diff-55f04b18703dd2d483d06741754106509cddb007abaeb0cac3b33ab6f166534aR443